src/main/scala/edu/arizona/sista/learning/Dataset.scala

```diff
@@ -89,26 +92,86 @@ class BVFDataset[L, F] (
     c
   }
 
-  def countFeatures(fs:ArrayBuffer[Array[Int]]):Counter[Int] = {
+  def countFeatures(fs:ArrayBuffer[Array[Int]], threshold:Int):Set[Int] = {
     val counts = new Counter[Int]
     for(d <- fs) {
       for(f <- d) {
         counts.incrementCount(f)
       }
     }
-    counts
+    logger.debug("Total unique features before filtering: " + counts.size)
+
+    val passed = new mutable.HashSet[Int]()
```

Comment:
If you don't want to have to handle those conversions (i.e., mutable to immutable), you could go with something like this:

```scala
val passed: Set[Int] = for {
  f <- counts.keySet
  if(counts.getCount(f) >= threshold)
} yield f
```